plugins/cms/src/cms.cpp

```diff
@@ -0,0 +1,267 @@
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:  The above copyright notice and this
+ * permission notice (including the next paragraph) shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "cms.h"
+#include "cmsMultispectralLink.h"
+
+#include <cereal/archives/json.hpp>
+#include <cereal/cereal.hpp>
+
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <numeric>
+#include <system_error>
+#include <utility>
+
+#include "neurala/plugin/PluginArguments.h"
+#include "neurala/plugin/PluginBindings.h"
+#include "neurala/plugin/PluginErrorCallback.h"
+#include "neurala/plugin/PluginManager.h"
+#include "neurala/plugin/PluginStatus.h"
+#include "neurala/utils/Version.h"
+#include "neurala/video/dto/CameraInfo.h"
+
+namespace
+{
+int
+exitHere()
+{
+	return 0;
+}
+
+std::unique_ptr<cmsMultiSpectralLink> multiSpectralLink;
+std::mutex cmsLock;
+std::condition_variable cmsCV;
+
+neurala::dto::ImageMetadata cmsMetadata;
+neurala::dto::ImageView cmsFrame;
+
+bool waitingForFrame;
+
+void
+cmsEventHandler()
+{
+	{
+		std::scoped_lock guard(cmsLock);
+
+		if (waitingForFrame)
+		{
+			waitingForFrame = false;
+			multiSpectralLink->getCmsImages(0)->setImageRaw(); //set the image raw
+			multiSpectralLink->getCmsImages(0)->calcImageMultisprectralCube(); //computing the multispectral cube
+			multiSpectralLink->getCmsImages(0)->calcImageColor(); //computing of the RGB and BGR images
+
+			const auto display = multiSpectralLink->getCmsImages(0)->getImageBGRDisplay();
+
+			cmsMetadata = neurala::dto::ImageMetadata("uint8", display->width, display->height, "BGR", "interleaved", "topLeft");
+			cmsFrame = neurala::dto::ImageView(cmsMetadata, display->imageData);
+		}
+	}
+
+	cmsCV.notify_all();
+}
+
+template<class Type, class Archive>
+auto deserialize(Archive& archive, const char* property) {
+	Type value;
+	archive(cereal::make_nvp(property, value));
+	return value;
+}
+
+template<class Archive>
+void load(Archive& archive, cmsMultiSpectralLink& cms)
+{
+	// cms.getCmsCamera()->setBlackLevel(deserialize<int>(archive, "blackLevel"));
+	// cms.getCmsCamera()->setExposure(deserialize<double>(archive, "exposureTime"));
```

Comment:
Reason for commenting / uncommenting?

Reply:
We agreed to set only frame rate and gain in our code and let the rest be controlled by the user, because these two parameters are for whatever reason the only two that don't get preserved across activations of the camera. The rest were commented out just as a record of what options there originally were, though I guess IntelliSense could point those out for me. I can remove them.

Reply:
Either remove or comment the section, explaining what can be achieved by uncommenting.